sshd/src/test/scala/ammonite/repl/sshd/SshTestingTraits.scala

```diff
@@ -0,0 +1,131 @@
+package ammonite.repl.sshd
+
+import java.io._
+
+import ammonite.ops._
+import com.jcraft.jsch.{Channel, Session, JSch, UserInfo}
+import org.scalacheck.Gen
+
+import scala.annotation.tailrec
+import scala.util.Random
+
+trait SshTestingTraits {
+  def alphaNumStr = Gen.listOf(Gen.alphaNumChar).map(_.mkString)
+  def nonEmptyAlphaNumStr = Gen.nonEmptyListOf(Gen.alphaNumChar).map(_.mkString)
+
+  lazy val testUsername = "test"
+  lazy val testPassword = Random.alphanumeric.take(10).mkString
+  lazy val testUser = (testUsername, testPassword)
+
+  def genCreds = for {
+    login ← nonEmptyAlphaNumStr
+    password ← alphaNumStr
+  } yield (login, password)
+
+  def genCredsPair = for {
+    creds1 ← genCreds
+    creds2 ← genCreds
+  } yield (creds1, creds2)
+
+  def genNonMatchingCredsPair = genCredsPair.suchThat(pair ⇒ pair._1 != pair._2)
+
+  def uniqueUsersWithOneOfThem = for {
+    users:List[(String, String)] ← Gen.nonEmptyListOf(genCreds)
+    noDuplicatedLogins = users.groupBy(_._1).values.map(_.head).toList
+    someUserIndex ← Gen.chooseNum(0, noDuplicatedLogins.size - 1)
+  } yield (noDuplicatedLogins, noDuplicatedLogins(someUserIndex))
+
+  def withTmpDirectory[T](block:Path ⇒ T):T = {
+    lazy val tmpDir = Path(Path.makeTmp)
+    try {
+      block(tmpDir)
+    } finally {
+      rm! tmpDir
+    }
+  }
+
+  def testSshServer(user:(String, String), shell: SshServer.RemoteShellServer)(implicit dir:Path) = {
+    SshServer(SshServerConfig(port = 0, username = user._1, password = user._2, ammoniteHome = dir), shell)
+  }
+
+  def withTestSshServer[T](user:(String, String), shell: SshServer.RemoteShellServer)
```

Comment:
As far as I can tell, `SshServer.RemoteShellServer` never gets its callback arguments ever used. This should take a `Function0` or a by-name param